### PR TITLE
Soc:arm: Add support for stm32l475(Alientek Pandora board)

### PR DESCRIPTION
**Pull request by TrekMax**
Drives that have been verified at present
- GPIO
- PWM
- LCD_ST7789
- QSPI_FLASH_W25Q128

### CODE REVIEW

File: boards/arm/pandora_stm32l475/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+Overview
+********
+
+The STM32L475 Pandora board features an ARM Cortex-M4 based STM32L475VE MCU
+with a wide range of connectivity support and configurations. Here are
+some highlights of the STM32L475 Pandora board:
+
+
+- STM32L476VGT6 microcontroller featuring 1 Mbyte of Flash memory, 128 Kbytes of RAM in LQFP100 package
```
Comment (FRASTM) [inline boards/arm/pandora_stm32l475/doc/index.rst:14]: maybe **STM32L475Vx** ?  but stm32l475VE has 512KB of Flash

File: boards/arm/pandora_stm32l475/pandora_stm32l475.dts [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+	};
+};
+
+&quadspi {
+	pinctrl-0 = <&quadspi_clk_pe10 &quadspi_ncs_pe11
+			&quadspi_bk1_io0_pe12 &quadspi_bk1_io1_pe13
+			&quadspi_bk1_io2_pe14 &quadspi_bk1_io3_pe15>;
+	pinctrl-names = "default";
+	// dmas = <&dma1 5 5 0x0000>;
+	// dma-names = "tx_rx";
```
Comment (FRASTM) [inline boards/arm/pandora_stm32l475/pandora_stm32l475.dts:142]: just remove lines if DMA is not used

File: soc/arm/st_stm32/stm32l4/Kconfig.soc [hunk 3]
```diff
@@ -13,6 +13,9 @@ choice
 config SOC_STM32L476XX
 	bool "STM32L476XX"
 
+config SOC_STM32L475XX
+	bool "STM32L475XX"
+
```
Comment (FRASTM) [inline soc/arm/st_stm32/stm32l4/Kconfig.soc:18]: already present at line 46
```
config SOC_STM32L475XX
	bool "STM32L475XX"
```

**Review by FRASTM — CHANGES_REQUESTED** [review]

**Review by FRASTM — CHANGES_REQUESTED** [review]

File: boards/arm/pandora_stm32l475/pandora_stm32l475.dts (hunk 2 above)
Comment (QinYUN575) [reply boards/arm/pandora_stm32l475/pandora_stm32l475.dts]: Revised, thank you

File: soc/arm/st_stm32/stm32l4/Kconfig.soc (hunk 3 above)
Comment (QinYUN575) [reply soc/arm/st_stm32/stm32l4/Kconfig.soc]: 🆗

File: dts/arm/st/l4/stm32l475Xe.dtsi [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
```
[the remark this answers is not in the document]
Comment (QinYUN575) [reply dts/arm/st/l4/stm32l475Xe.dtsi]: 🆗

**Review by FRASTM — DISMISSED** [review]

File: drivers/display/display_st7789v.c [hunk 5]
```diff
@@ -363,6 +364,17 @@ static int st7789v_init(const struct device *dev)
...
+		if (!device_is_ready(config->bl_ctrl_gpio.port)) {
+			LOG_ERR("Backlight GPIO device not ready");
+			return -ENODEV;
+		}
+
+		if (gpio_pin_configure_dt(&config->bl_ctrl_gpio, GPIO_OUTPUT_INACTIVE)) {
+			LOG_ERR("Couldn't configure Backlight pin");
+			return -EIO;
+		}
+	}
```
Comment (jfischer-no) [inline drivers/display/display_st7789v.c:377]: -2, not like this and not in this PR.

**Review by jfischer-no — DISMISSED** [review]

File: drivers/display/display_st7789v.c (hunk 5 above)
Comment (QinYUN575) [reply drivers/display/display_st7789v.c:377]: -ENOENT?

Comment (jfischer-no) [reply drivers/display/display_st7789v.c:377]: Changes in [drivers/display/display_st7789v.c](https://github.com/zephyrproject-rtos/zephyr/pull/55577/commits/8fec1e03a69a7300acd1b736e0e726500b8d6c81#diff-73d589a4a3df091a005e026fb9c14c1ee3d878c8746f02cd4ee592aacc075e49)  are not correct and do not belong in this PR.

Comment (QinYUN575) [reply drivers/display/display_st7789v.c]: Thanks, removed commits not related to this PR

**Comment by QinYUN575** [comment]
Fixed the lack of link field in the document, which caused the ci detection to fail

**Review by FRASTM — COMMENTED** [review]
still compliance to fix and check the url link in the index.rst

else LGTM

**Comment by QinYUN575** [comment]
Modify the commit message according to the "Commit Message Guidelines"

**Comment by MaureenHelm** [comment]
@QinYUN575 please address the failed compliance checks

**Comment by erwango** [comment]
@QinYUN575 See https://github.com/zephyrproject-rtos/zephyr/actions/runs/4711141843/jobs/8369955856?pr=55577#step:9:1

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/pandora_stm32l475/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
```
[the remark this answers is not in the document]
Comment (erwango) [reply boards/arm/pandora_stm32l475/doc/index.rst:107]: Would you mind completing this table, while you're at ?

[the remark this answers is not in the document]
Comment (QinYUN575) [reply boards/arm/pandora_stm32l475/doc/index.rst:107]: Hold on, I'm editing now.

**Comment by erwango** [comment]
Can you try: `boards: arm: support board Pandora_STM32L475` the following commit title, please note use of spaces (` `).

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**